client/internal/internal.go

```diff
@@ -0,0 +1,71 @@
+package internal
+import (
+    "github.com/derekparker/delve/proctl"
+    "github.com/derekparker/delve/goreadline"
+    "fmt"
+    "io"
+    "strings"
+    "os"
+    sys "golang.org/x/sys/unix"
+)
+const HistoryFile string = ".dbg_history"
+
+
+func HandleExit(dbp *proctl.DebuggedProcess, status int, shouldConfirm bool) {
+    errno := goreadline.WriteHistoryToFile(HistoryFile)
+    if errno != 0 {
+        fmt.Println("readline:", errno)
+    }
+
+    var answer string
+    if shouldConfirm {
+        prompt := "Would you like to kill the process? [y/n]"
+        answerp := goreadline.ReadLine(&prompt)
+        if answerp == nil {
+            Die(2, io.EOF)
+        }
+        answer = strings.TrimSuffix(*answerp, "\n")
+    }
+    for _, bp := range dbp.HWBreakPoints {
```

Comment:
I think it might make sense to move this breakpoint clearing into a method off of `DebuggedProcess` - something like `ClearAllBreakpoints`. That would remove duplication across clients, and would probably move this function back into the cli client, with another being necessary for the ws client.